lib/leanplum_api/api.rb

```diff
@@ -1,58 +1,53 @@
 module LeanplumApi
   class API
+    EXPORT_PENDING = 'PENDING'.freeze
+    EXPORT_RUNNING = 'RUNNING'.freeze
+    EXPORT_FINISHED = 'FINISHED'.freeze
     extend Gem::Deprecate
 
     class LeanplumValidationException < RuntimeError; end
 
-    EXPORT_PENDING = 'PENDING'
-    EXPORT_RUNNING = 'RUNNING'
-    EXPORT_FINISHED = 'FINISHED'
-
     def initialize(options = {})
       fail 'LeanplumApi not configured yet!' unless LeanplumApi.configuration
     end
 
     def set_user_attributes(user_attributes, options = {})
-      track_multi(nil, user_attributes, options)
+      track_multi(user_attributes: user_attributes, options: options)
+    end
+
+    def set_device_attributes(device_attributes, options = {})
+      track_multi(device_attributes: device_attributes, options: options)
     end
 
     def track_events(events, options = {})
-      track_multi(events, nil, options)
+      track_multi(events: events, options: options)
     end
 
-    # This method is for tracking events and/or updating user attributes at the same time, batched together like
-    # leanplum recommends.
-    # Set the :force_anomalous_override option to catch warnings from leanplum about anomalous events and force them to
-    # not be considered anomalous.
-    def track_multi(events = nil, user_attributes = nil, options = {})
-      request_data = Array.wrap(user_attributes).map { |h| build_user_attributes_hash(h) } +
-                     Array.wrap(events).map { |h| build_event_attributes_hash(h, options) }
+    # This method is for tracking events and/or updating user attributes
+    # at the same time, batched together like leanplum recommends.
+    # Set the :force_anomalous_override option to catch warnings from leanplum
+    # Ôabout anomalous events and force them to not be considered anomalous.
+    def track_multi(events: nil, user_attributes: nil, device_attributes: nil, options: {})
```

Comment:
though maybe i just don't understand this newfangled syntax?  can i still call 
```
track_multi(events, user_attributes)
```
without the keys?  or does it have to be
```ruby
track_multi(events: events, user_attributes: user_attributes)
```

Reply:
It requires a change to the caller. I'm going to back this change out as it would require more upstream changes. Its a newer way to call methods with variable interface calls. But I'll keep to the existing pattern for this change.

Reply:
Hmmm... I do need to add device_attributes to the interface. I did not like the idea of forcing the use of options or type check to distinguish between user and device attributes.

Reply:
yeah i'm cool with this change, actually... having 3 hash arguments + an options hash is kind of gross.  needs to go to 4.0 anyways so have at it.

Reply:
Its actually 2 hashes. or specifically a hash with a hash as value for options key. this is a goods addition to ruby for reducing the need for positional arguments and using keys.